src/hybrid_reader.c

```diff
@@ -399,14 +408,14 @@ IndexIterator *NewHybridVectorIterator(HybridIteratorParams hParams) {
   hi->runtimeParams = hParams.qParams;
   hi->scoreField = hParams.vectorScoreField;
   hi->base.isValid = 1;
-  hi->list = NULL;
+  memset(&hi->list, 0, sizeof(hi->list));
   hi->iter = NULL;
   hi->topResults = NULL;
   hi->returnedResults = NULL;
   hi->numIterations = 0;
   hi->ignoreScores = hParams.ignoreDocScore;
-  hi->timeoutCb = TimedOut_WithCtx;
   hi->timeoutCtx = (TimeoutCtx){ .timeout = hParams.timeout, .counter = 0 };
+  hi->runtimeParams.timeoutCtx = &hi->timeoutCtx;
```

Comment:
Why do you need to store `hi->timeoutCtx` ptr twice?

Reply:
Its not saving the pointer twice. the `timeoutCtx` in the hybrid iterator is the actual context struct, and we pass its address in the query params